.github/workflows/windows.yml

```diff
@@ -3,104 +3,27 @@ name: Windows builds
-      CCACHE_MAXSIZE:  "64M"
-      CCACHE_COMPRESS: "true"
-    steps:
-      - uses:  actions/checkout@v2
-      - name:  Get Date
-        id:    get-date
-        shell: bash
-        run:   echo "::set-output name=date::$(date +%Y-%W)"
-      - uses:  actions/cache@v1
-        id:    cache-msys2
-        with:
-          path: 'C:/tools/msys64'
-          key: msys2-${{ matrix.conf.bits }}-${{ steps.get-date.outputs.date }}
-      - name:  Install MSYS2
-        if:    steps.cache-msys2.outputs.cache-hit != 'true'
-        run:   choco install msys2 --no-progress
-      - name:  Install C++ compiler and libraries
-        if:    steps.cache-msys2.outputs.cache-hit != 'true'
-        shell: python scripts\msys-bash.py {0}
-        run:   ./scripts/list-build-dependencies.sh -m msys2 -c clang -b ${{ matrix.conf.bits }} | xargs pacman -S --noconfirm
-      - name:  Shrink MSYS2 for cache
-        if:    steps.cache-msys2.outputs.cache-hit != 'true'
-        shell: python scripts\msys-bash.py {0}
-        run:   ./.github/scripts/shrink-msys2.sh
-      - name:  Prepare compiler cache
-        id:    prep-ccache
-        shell: bash
-        run: |
-          mkdir -p "${CCACHE_DIR}"
-          echo "::set-output name=dir::$CCACHE_DIR"
-          echo "::set-output name=today::$(date -I)"
-          echo "::set-output name=yesterday::$(date --date=yesterday -I)"
-      - uses:  actions/cache@v1
-        id:    cache-ccache
-        with:
-          path: ${{ steps.prep-ccache.outputs.dir }}
-          key:  ccache-msys2-${{ matrix.conf.compiler }}-${{ matrix.conf.bits }}-${{ steps.prep-ccache.outputs.today }}
-          restore-keys: |
-            ccache-msys2-${{ matrix.conf.compiler }}-${{ matrix.conf.bits }}-${{ steps.prep-ccache.outputs.yesterday }}
-      - name:  Log environment
-        shell: python scripts\msys-bash.py {0}
-        run:   ./scripts/log-env.sh
-      - name:  Build
-        shell: python scripts\msys-bash.py {0}
-        run: |
-          # The following two MSYSTEM variables are used by auto-tools to determine host
-          # architecture so we override them to match the current build type (32-bit or 64-bit)
-          export MSYSTEM_CARCH=${{ matrix.conf.arch }}
-          export MSYSTEM_CHOST=${{ matrix.conf.arch }}-pc-msys
-          ./scripts/build.sh --compiler ${{ matrix.conf.compiler }} --build-type Debug --bin-path /mingw${{ matrix.conf.bits }}/bin
-      - name:  Summarize warnings
-        env:
-          MAX_WARNINGS: ${{ matrix.conf.max_warnings }}
-        run:   python scripts\count-warnings.py build.log
-
   build_windows_vs:
-    name: MSVC 32-bit
+    name: ${{ matrix.conf.name }}
     runs-on: windows-latest
     if: github.event_name != 'pull_request' || contains('dreamer,kcgen,ant-222', github.actor) == false
     strategy:
       matrix:
-        type: [Debug]
+        conf:
+          - name: MSVC 32-bit
+            vs-platform: x86
+            vcpkg-triplet: x86-windows
```

Comment:
That's handy; I didn't know about triplets! (https://github.com/microsoft/vcpkg/blob/master/docs/users/triplets.md).

Reply:
Yeah, the introduction of this was quite a good idea by vcpkg developers :) Quite easy to use. Rust also decided on a very similar scheme to name the platform support:

https://forge.rust-lang.org/release/platform-support.html#tier-1